pkg/network/vmispec/interface.go

```diff
@@ -0,0 +1,62 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2021 Red Hat, Inc.
+ *
+ */
+
+package vmispec
+
+import (
+	v1 "kubevirt.io/api/core/v1"
+)
+
+func PodNetworkHasOnlyMasqueradeBindingInterfaces(networks []v1.Network, ifaces []v1.Interface) bool {
+	if podNetwork := lookupPodNetwork(networks); podNetwork != nil {
+		podInterfaces := filterInterfacesByNetwork(ifaces, podNetwork)
+		return len(filterMasqueradeBindingInterfaces(podInterfaces)) == len(podInterfaces)
+	}
+	return true
+}
+
+func filterInterfacesByNetwork(ifaces []v1.Interface, network *v1.Network) []v1.Interface {
+	var networkIfaces []v1.Interface
+	for _, iface := range ifaces {
+		if iface.Name == network.Name {
+			networkIfaces = append(networkIfaces, iface)
+		}
+	}
+	return networkIfaces
+}
+
+func filterMasqueradeBindingInterfaces(ifaces []v1.Interface) []v1.Interface {
+	var masqIfaces []v1.Interface
+	for _, iface := range ifaces {
+		if iface.Masquerade != nil {
+			masqIfaces = append(masqIfaces, iface)
+		}
+	}
+	return masqIfaces
+}
+
+func FilterSRIOVInterfaces(ifaces []v1.Interface) []v1.Interface {
+	var sriovIfaces []v1.Interface
+	for _, iface := range ifaces {
+		if iface.SRIOV != nil {
+			sriovIfaces = append(sriovIfaces, iface)
+		}
+	}
+	return sriovIfaces
```

Comment:
There is a slight performance impact here with comparison to the previous `validateSRIOVInterfacesForMigration()` function which ended the loop as soon as an SRIOV interface was found.

It could be negligible, just wanted to point that out.

Reply:
> This is processed in memory on a limited number of entries, therefore the performance is negligible up to none.
The reasoning for using this instead of the previous one is the clarity and generalization of the steps done. For me, it is much simpler to understand what is done.

Agree 100%. Thanks.